pkg/manager/manager.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2019 The KubeMacPool Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package manager
+
+import (
+	"flag"
+	"fmt"
+	"k8s.io/client-go/rest"
+	"net"
+	"os"
+	"os/signal"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"time"
+
+	"github.com/go-logr/logr"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	"github.com/K8sNetworkPlumbingWG/kubemacpool/pkg/controller"
+	poolmanager "github.com/K8sNetworkPlumbingWG/kubemacpool/pkg/pool-manager"
+	"github.com/K8sNetworkPlumbingWG/kubemacpool/pkg/webhook"
+)
+
+var log logr.Logger
+
+type KubeMacPoolManager struct {
+	clientset                *kubernetes.Clientset
+	config                   *rest.Config
+	metricsAddr              string
+	continueToRunManager     bool
+	restartChannel           chan struct{}
+	kubevirtInstalledChannel chan struct{}
+	stopSignalChannel        chan os.Signal
+}
+
+func NewKubeMacPoolManager() *KubeMacPoolManager {
+	kubemacpoolManager := &KubeMacPoolManager{continueToRunManager: true,
+		restartChannel:           make(chan struct{}),
+		kubevirtInstalledChannel: make(chan struct{}),
+		stopSignalChannel:        make(chan os.Signal, 1)}
+
+	signal.Notify(kubemacpoolManager.stopSignalChannel, os.Interrupt, os.Kill)
+
+	flag.StringVar(&kubemacpoolManager.metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
```

Comment:
Flags should be handled in main(). This function should receive native Go arguments passed from main. That way we split the application from its "front-end". Logger should be initialized in the main too. It depends on user of this application how logging should be handled.